test/unit/conversations_page_test.dart

```diff
@@ -16,5 +16,19 @@ void main() {
       vm.add(item: instance);
       expect(inject_list, [instance]);
     });
+    test('test ConversationViewModel populateWith', () {
+      var inject_list = <ConversationItem>[];
+      var vm = ConversationsViewModel(inject_list); // start with empty list
+      var instance = ConversationItem(
+          conversationId: 'abc123',
+          uids: ['123', '456'],
+          displayNames: ['Leon', 'Noel'],
+          photoURLs: ['https://url1', 'https://url2']);
+      var alternate_list = <ConversationItem>[];
+      alternate_list.add(instance);
+
+      vm.populateWith(alternate_list);
+      expect(inject_list, alternate_list);
```

Comment:
I was kind of surprised this works as Dart by default uses identity equivalence so I looked into it a little.  

The arguments to the [expect function](https://pub.dev/documentation/test_api/latest/test_api/expect.html) are `actual` and `matcher` but if `matcher` is just a value (like we are doing here) then it is "wrapped in an equals matcher", ie. the [equals function](https://pub.dev/documentation/matcher/latest/matcher/equals.html).  

The `equals function` tests for equality using == on the expected value, but *for Iterables and Maps, this will recursively match the elements*. 

So we get state based equivalence for collections when using `expect`... good to know!